packages/dm-core/src/domain/Tree.tsx

```diff
@@ -179,9 +189,14 @@ export class TreeNode {
     this.isDataSource = isDataSource
     this.entity = entity
     this.name = name || entity?.name
-    this.type = Object.keys(entity).length
-      ? entity.type
-      : attribute.attributeType
+    if (Array.isArray(entity)) {
+      this.type = attribute.attributeType
+    } else {
+      this.type = Object.keys(entity).length
+        ? entity.type
+        : attribute.attributeType
+    }
```

Comment:
```suggestion
    this.type = entity && Object.hasOwn(entity, "type")
      ? entity.type
      : attribute.attributeType
```
would also work. More robust I think. If entity is an array, it will be converted to an object before testing (ie ["a", "g"] -> {"0": "a", "1": "g"}) but the keys will never be "type"